packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1234,6 +1234,15 @@ function updateSuspenseComponent(
           NoWork,
           null,
         );
+
+        if (enableProfilerTimer && workInProgress.mode & ProfileMode) {
+          // Fiber treeBaseDuration must be at least as large as the sum of children treeBaseDurations.
+          // Otherwise the profiler's onRender metrics will be off,
+          // and the DevTools Profiler flamegraph will visually break as well.
+          primaryChildFragment.treeBaseDuration =
+            currentPrimaryChild.treeBaseDuration;
```

Comment:
Need to transfer from the siblings, too.

Reply:
Good catch. Fixed and updated tests to cover this via 85c5831